sharkfuser/samples/convolution/conv_fprop_grouped_nchw_kcrs.cpp

```diff
@@ -0,0 +1,120 @@
+    auto graph = std::make_shared<Graph>();
+    graph->setName("conv_fprop_grouped_sample_nchw_kcrs_1x1_nopad");
+    graph->setIODataType(DataType::Half).setComputeDataType(DataType::Float);
+
+    auto X = graph->tensor(TensorAttr()
+                               .setName("image")
+                               .setDim({n, c, h, w})
+                               .setStride({c * h * w, h * w, w, 1})); // NCHW
+
+    auto W = graph->tensor(TensorAttr()
+                               .setName("filter")
+                               .setDim({k, fc, r, s})
+                               .setStride({fc * r * s, r * s, s, 1})); // KCRS
+
+    auto convAttr = ConvFPropAttr()
+                        .setPadding({0, 0})
+                        .setStride({1, 1})
+                        .setDilation({1, 1})
+                        .setName("conv_fprop");
+
+    auto Y = graph->convFProp(X, W, convAttr);
+    Y->setOutput(true);
+
+    // Validate, infer missing properties
+    FUSILLI_REQUIRE_OK(graph->validate());
+
+    // Compile
+    FUSILLI_REQUIRE_OK(graph->compile(handle, /*remove=*/true));
+
+    return std::make_tuple(graph, X, W, Y);
+  };
+
+  // Parameterize sample by backend and create device-specific handles.
+  std::shared_ptr<Handle> handlePtr;
+  SECTION("cpu backend") {
+    handlePtr = std::make_shared<Handle>(
+        FUSILLI_REQUIRE_UNWRAP(Handle::create(Backend::CPU)));
+  }
+#ifdef FUSILLI_ENABLE_AMDGPU
+  SECTION("amdgpu backend") {
+    handlePtr = std::make_shared<Handle>(
+        FUSILLI_REQUIRE_UNWRAP(Handle::create(Backend::AMDGPU)));
+  }
+#endif
+  Handle &handle = *handlePtr;
+
+  // Build graph for the given handle (device), validate and compile it.
+  auto [graph, X, W, Y] = build_new_graph(handle);
+
+  // Allocate input buffer.
+  auto xBuf = std::make_shared<Buffer>(FUSILLI_REQUIRE_UNWRAP(Buffer::allocate(
+      handle,
+      /*shape=*/castToSizeT(X->getPhysicalDim()),
+      /*data=*/std::vector<half>(X->getVolume(), half(1.0f)))));
+
+  // Allocate weight buffer.
+  auto wBuf = std::make_shared<Buffer>(FUSILLI_REQUIRE_UNWRAP(Buffer::allocate(
+      handle,
+      /*shape=*/castToSizeT(W->getPhysicalDim()),
+      /*data=*/std::vector<half>(W->getVolume(), half(1.0f)))));
+
+  // Allocate output buffer.
+  auto yBuf = std::make_shared<Buffer>(FUSILLI_REQUIRE_UNWRAP(Buffer::allocate(
+      handle,
+      /*shape=*/castToSizeT(Y->getPhysicalDim()),
+      /*data=*/std::vector<half>(Y->getVolume(), half(0.0f)))));
+
+  // Create variant pack.
+  const std::unordered_map<std::shared_ptr<TensorAttr>, std::shared_ptr<Buffer>>
+      variantPack = {
+          {X, xBuf},
+          {W, wBuf},
+          {Y, yBuf},
+      };
+
+  // Execute graph once.
+  FUSILLI_REQUIRE_OK(graph->execute(variantPack));
+
+  // Read output buffers.
+  std::vector<half> result;
+  FUSILLI_REQUIRE_OK(yBuf->read(handle, result));
+  for (auto val : result)
+    REQUIRE(val == half(16.0f));
```

Comment:
probably a good idea to use a constexpr for the expected value 16.0f (so it's clear how it was arrived at)...